pkg/containerimagebuilderpusher/kaniko.go

```diff
@@ -384,49 +365,61 @@ func (k *Kaniko) configureECRInitContainerAndMount(buildOptions *BuildOptions, k
 		"|| if [ $? -eq 254 ]; then echo 'Ignoring repository already exits'; else exit $?; fi",
 		buildOptions.RepoName,
 		k.resolveAWSRegionFromECR(buildOptions.RegistryURL))
-	kanikoJobSpec.Spec.Template.Spec.InitContainers = append(kanikoJobSpec.Spec.Template.Spec.InitContainers,
-		v1.Container{
-			Name:  "create-repo",
-			Image: k.builderConfiguration.AWSCLIImage,
-			Command: []string{
-				"/bin/sh",
-			},
-			Args: []string{
-				"-c",
-				createRepoCommand,
-			},
+	initContainer := v1.Container{
+		Name:  "create-repo",
+		Image: k.builderConfiguration.AWSCLIImage,
+		Command: []string{
+			"/bin/sh",
+		},
+		Args: []string{
+			"-c",
+			createRepoCommand,
+		},
+	}
 
-			// mount the credentials file to /tmp for permissions reasons
-			Env: []v1.EnvVar{
-				{
-					Name:  "AWS_SHARED_CREDENTIALS_FILE",
-					Value: "/tmp/credentials",
-				},
+	if k.builderConfiguration.AWSSecretName != "" {
```

Comment:
Is this secret different than `buildOptions.SecretName` (which is used in line 334)?